### PR TITLE
Server engine 2

### PR DESCRIPTION
**Pull request by notzippy**
Minimum changes for server engine to work

### CODE REVIEW

File: server-engine/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+These are some of the server engines built by the Revel team.
+In order to use an engine you must import the module and set
+the configuration option `server.engine` equal to the engine you want.
+You may define your own engine by implementing `revel.ServerEngine` and
+registering it with revel in an `init` 
+function `revel.RegisterServerEngine(NAME, <function to return your server instance>)`. See
+`revel.GOHttpServer` for an example
+
+
+###App.conf
```
Comment (brendensoares) [inline server-engine/README.md]: space, also app.conf?

File: server-engine/fasthttp/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#FastHTTP
```
Comment (brendensoares) [inline server-engine/fasthttp/README.md]: space

File: server-engine/fasthttp/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+This module is the for [FastHTTP](https://github.com/valyala/fasthttp) server engine.
+It does not support WebSockets.
+
+###App.conf
```
Comment (brendensoares) [inline server-engine/fasthttp/README.md]: space, app

File: server-engine/fasthttp/README.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+- **server.engine** You must set this to `fasthttp` in order to use this server engine
+
+###Other Notes
```
Comment (brendensoares) [inline server-engine/fasthttp/README.md]: space

File: server-engine/fasthttp/README.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+All features from supported by a regular HTTP engine is supported by this server engine.
```
Comment (brendensoares) [inline server-engine/fasthttp/README.md]: wording?

File: server-engine/fasthttp/README.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+Memory usage is decreased because this engine makes reuse of allocated structures to
+handle requests. This should increase overall runtime performance and throughput.
```
Comment (brendensoares) [inline server-engine/fasthttp/README.md]: "by avoiding memory allocations". Also, should we mention request pool concerns here?

File: server-engine/README.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
+# Server Engines
+These are some of the server engines built by the Revel team.
+In order to use an engine you must import the module and set
+the configuration option `server.engine` equal to the engine you want.
+You may define your own engine by implementing `revel.ServerEngine` and
+registering it with revel in an `init` 
+function `revel.RegisterServerEngine(NAME, <function to return your server instance>)`. See
+`revel.GOHttpServer` for an example
```
Comment (brendensoares) [inline server-engine/README.md]: `GO` to `Go`

File: server-engine/fasthttp/README.md (hunk 6 above)
Comment (notzippy) [inline server-engine/fasthttp/README.md]: re pool concerns: I think that topic is better for the manual

**Pull request merged**